### PR TITLE
Fixed all UP007 ruff codestyle violations

### PR DESCRIPTION
**Pull request by Bryce-3D**
Fixed all UP007 ruff codestyle violations https://docs.astral.sh/ruff/rules/non-pep604-annotation/.

Also correspondingly removed all unneeded copies of `from typing import Optional, Union`.

Please merge the other pull request first at https://github.com/lunaflight/courserekt/pull/8 since that happened before this, though that change is extremely minor so just merging this one directly probably works too ig?

### CODE REVIEW

File: .gitignore
Comment (lunaflight) [inline .gitignore:1]: These changes seem to already be covered by the other pull request.
Did you mean to make only 1 pull request, or 2 pull requests - one to fix `.gitignore` only, and the other to fix more style errors?

**Review by lunaflight — CHANGES_REQUESTED** [review]
Looks good to me, except the comment on `style-errors.txt` similar to #8.

File: .gitignore
Comment (Bryce-3D) [reply .gitignore:1]: Two pull requests was my intention, but this single pull request is fine too. I've also corrected style-errors.txt now.

**Review by lunaflight — APPROVED** [review]
Looks good to me :3

**Pull request merged**